File: projects/04/clojure/src/factorial.clj
;;;; factorial.clj calculates the factorial
;;;;
;;;; Input: number
;;;; Precondition: number is integer
;;;; Output: factorial of given number
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed by: Jacob Brink
;;;; Date: 2/28/2020
;;;; Project: 4
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns factorial)                                 ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; calculateFactorial() calculates a factorial
;;; Receive: number
;;; Precondition: number is integer
;;; Output: factorial of number
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;


(defn calculateFactorial [number]
    (loop [i 2 answer 1]
      (if (<= i number)
         (recur (inc i) (* answer i))
         answer
         )
      )    
  )

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; -main is a 'driver' for the 
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(defn -main []
 (println "Enter a number: ") (flush)
 (let
    [ number (double (read)) ] 
   (println)
   (printf "The factorial of %s is %s" number (calculateFactorial number))
 )
)

File: projects/04/factorial.rb
#! /usr/bin/ruby
# factorial.rb calculates the factorial of a given number
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by: Jacob Brink
# Date: 2/28/2020
# Project: 4
##########################################################

# Input: Number
# Precondition: Number is integer
# Output: factorial of Number

def calcFactorial(number)
  answer = 1
  for i in 2..number
    answer *= i
  end
  return answer
end

if __FILE__ == $0
   print "Enter a number: \n"
   number = gets.chomp.to_f
   print "The factorial of #{number} is #{calcFactorial(number)}\n"
end
